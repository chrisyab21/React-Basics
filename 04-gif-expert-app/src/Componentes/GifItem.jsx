
import PropTypes from 'prop-types'
import React from 'react'

export const GifItem = ({title, id, url}) => {

  return (
   

   <div className='card'>

    <img src={url} alt={title}></img>
    <p>{title}</p>

   </div>

   
  )
}


GifItem.propTypes = {


   title: PropTypes.string.isRequired,
   url: PropTypes.string.isRequired

}


GifItem.defaultProps = {

  title: "Aqui va el titulo",
  url: "Aqui va la url"
}

