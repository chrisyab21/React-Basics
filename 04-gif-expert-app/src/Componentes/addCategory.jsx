import React, { useState } from 'react'

import PropTypes from 'prop-types'

export const AddCategory = ({onAddCategory}) => {


    const [inputValue , setInputValue] = useState('');


    const writeOnChange = (event) =>{

       
        setInputValue(event.target.value);
       // console.log(event.target.value);
    }

    const onSubmit = (event) =>{
       
        event.preventDefault();
    //    console.log(inputValue);

        if(inputValue.trim().length <=1){

            return;
        }

        // onAddCategory((categories) => [inputValue, ...categories]);

        onAddCategory(inputValue.trim());
        
        setInputValue('');
        
    }

  return (

    <form onSubmit={(event)=> onSubmit(event)} aria-label='form'>

    <input 
    type='text'
    placeholder='Buscar Gifs'
    value={inputValue}
    onChange={(event) => writeOnChange(event)}
    //Segunda forma de escribir la linea de forma mas directa.
    //onChange={writeOnChange}
    >
    </input>

    </form>
  )
}


AddCategory.propTypes = {

  onAddCategory: PropTypes.func.isRequired,


}
