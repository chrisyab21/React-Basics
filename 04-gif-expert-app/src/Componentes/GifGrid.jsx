

import React, { useState, useEffect } from 'react'
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {


  const { Images, isLoading} = useFetchGifs(category);


 

  return (
    <>
        <h3>{category}</h3>


      {
          isLoading ? <h2>Cargando...</h2>
          : null

      }

       
       <ol className='card-grid'>
          {
            Images.map((datosImg) =>{

              return ( 
                
                <GifItem 
                key={datosImg.id} 
                	{...datosImg}
                ></GifItem>
                
                )

            } )

          }
        
       </ol>
    </>
  )
}




GifGrid.propTypes = {

category: PropTypes.string.isRequired,


}