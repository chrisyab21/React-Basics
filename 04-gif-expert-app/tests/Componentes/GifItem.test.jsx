

import { screen, render, fireEvent, getByText } from "@testing-library/react";
import { GifItem } from "../../src/Componentes/GifItem";

describe('Pruebas en GifItem', () => {
    

    test('Debe hacer match con el snapshot', () => {


        const title =  'Goku gafass';
        const url = 'https://media.giphy.com/media/125cxELHOpsLra/giphy.gif'
        const {container} = render(<GifItem title={title} url={url}></GifItem>);

        expect(container).toMatchSnapshot();
     });


     test('Debe contener una imagen con titulo y url solo una vez', () => { 

        const title =  'Goku gafass';
        const url = 'https://media.giphy.com/media/125cxELHOpsLra/giphy.gif'
         render(<GifItem title={title} url={url}></GifItem>);

        console.log(screen.debug());

        expect(screen.getAllByText(title).length).toBe(1);
        expect(screen.getAllByRole("img").length).toBe(1);
        
        });

        
     test('Debe mostrar la imagen con el URL y el ALT indicados', () => { 

        const title =  'Goku gafass';
        const url = 'https://media.giphy.com/media/125cxELHOpsLra/giphy.gif'
         render(<GifItem title={title} url={url}></GifItem>);

        console.log(screen.debug());
        
        expect(screen.getByRole('img').src).toBe(url);
        expect(screen.getByRole('img').alt).toBe(title);
        
        });



        


 });