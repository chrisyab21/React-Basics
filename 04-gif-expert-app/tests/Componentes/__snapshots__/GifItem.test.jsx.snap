// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku gafass"
      src="https://media.giphy.com/media/125cxELHOpsLra/giphy.gif"
    />
    <p>
      Goku gafass
    </p>
  </div>
</div>
`;
